force-app/testRecordGenerator/classes/TestFieldFunctions.cls

```diff
@@ -84,6 +84,14 @@ global class TestFieldFunctions {
         global String developerName;
 
         global Object getValue(String fieldName, Object value) {
+
+            if (sObjectType == null) {
+                sObjectType = parentGenerator.metadata.SObject__c;
+            }
+            if (developerName == null) {
+                sObjectType = (String) value;
```

Comment:
Neat idea, but should this line be 

```
developerName = (String) value;
```
? If so, then this change should probably come with a test 😛

Reply:
Whoops, good spot. I wondered how to test this without any valid record types in the package. It's probably worth setting up some mocking for NamedSObjectCache as I notice the NamedSObjectCacheTest only has coverage for the recordType() method.  Did you want me to set up some mocking?

Reply:
That's a good point, I created Issue #19 to support mocking, but let's get these changes merged in first.

Reply:
Hi @aidan-harding 

I did commit these items, https://github.com/aidan-harding/nebula-core/pull/11/commits

Do I need to do something else?

Reply:
That one is resolved and I've merged your changes, the open PR with comments from me is #10